letsencrypt/scripts/main.py

```diff
@@ -9,61 +9,100 @@
-                             "changes.")
-    parser.add_argument("-r", "--redirect", dest="redirect",
-                        action="store_const", const=True,
-                        help="Automatically redirect all HTTP traffic to HTTPS "
-                             "for the newly authenticated vhost.")
-    parser.add_argument("-n", "--no-redirect", dest="redirect",
-                        action="store_const", const=False,
-                        help="Skip the HTTPS redirect question, allowing both "
-                             "HTTP and HTTPS.")
-    parser.add_argument("-e", "--agree-tos", dest="eula", action="store_true",
-                        help="Skip the end user license agreement screen.")
-    parser.add_argument("-t", "--text", dest="use_curses", action="store_false",
-                        help="Use the text output instead of the curses UI.")
-    parser.add_argument("--test", dest="test", action="store_true",
-                        help="Run in test mode.")
+    add = parser.add_argument
+
+    add("-d", "--domains", metavar="DOMAIN", nargs="+")
+    add("-s", "--acme-server", "--server", default="letsencrypt-demo.org:443",
+        help="CA hostname (and optionally :port). The server certificate must "
+             "be trusted in order to avoid further modifications to the "
+             "client.")
+
+    add("-p", "--privkey", type=read_file,
+        help="Path to the private key file for certificate generation.")
+    add("-B", "--rsa-key-size", "--keysize", type=int, default=2048,
+        metavar="N", help="RSA key shall be sized N bits.")
+
+    add("-k", "--revoke", action="store_true", help="Revoke a certificate.")
+    add("-b", "--rollback", type=int, default=0, metavar="N",
+        help="Revert configuration N number of checkpoints.")
+    add("-v", "--view-config-changes", action="store_true",
+        help="View checkpoints and associated configuration changes.")
+    add("-r", "--redirect", action="store_true",
+        help="Automatically redirect all HTTP traffic to HTTPS for the newly "
+             "authenticated vhost.")
+
+    add("-e", "--agree-tos", dest="eula", action="store_true",
+        help="Skip the end user license agreement screen.")
+    add("-t", "--text", dest="use_curses", action="store_false",
+        help="Use the text output instead of the curses UI.")
+    add("--test", action="store_true", help="Run in test mode.")
+
+    # TODO: trailing slashes might be important! check and remove
+    add("--config-dir", default="/etc/letsencrypt/",
+        help="Configuration directory.")
+    add("--work-dir", default="/var/lib/letsencrypt/",
+        help="Working directory.")
+    add("--backup-dir", default="/var/lib/letsencrypt/backups/",
+        help="Configuration backups directory.")
+    add("--temp-checkpoint-dir",
+        default="/var/lib/letsencrypt/temp_checkpoint/",
+        help="Temporary checkpoint directory.")
+    add("--in-progress-dir",
```

Comment:
> (any ideas where?)
That is a good question.... of course it would be really nice to set it within the config object.  I know nothing about ConfArgParse other than the 4 minutes I just spent looking it up and that doesn't appear to be easy or straight-forward.  Solutions I have come up with... create a wrapper for confArgParse (probably confusing), deal with the extra directories being dynamic and confusing, eliminate the variables and simply compute them dynamically in the files we need them (my current top choice).

Rev tokens is only used in recovery_contact.py
cert_key_backup is used in revoker and client... though my refactoring should remove the directory from client.py
in-progress-dir and temp_checkpoint_dir are only used in reverter.py  

> I totally did not get your point about certificates,

No problem :) I am not sure I would have understood it if I came back to it in a week... I made several errors in my free-flowing quick thought.  I apologize for my bad English/writing.

Right now it seems as though both cert-dir and cert-path are fully specified.
"--cert-dir", default="/etc/letsencrypt/certs/"
"--cert-path", default="/etc/letsencrypt/certs/cert-letsencrypt.pem"

I believe I would find this confusing if I was quickly looking at the help saw cert-dir and specified it.  
I would run the program and my certificate is nowhere to be found, because I didn't specify the cert-path.

I think it is just a confusing situation... and I don't have a great solution.  Naturally, I think it makes more sense to say --cert-path=<path>  This does leave the chain unspecified though... which would be confusing when you are trying to install your certificate manually and you can't seem to find your chain certificate.

I think specifying cert-dir and keeping the files together is more important than naming the files individually.  Does the problem make sense?  Do you have any better ideas?